hollow/src/main/java/com/netflix/hollow/api/consumer/incubating/HollowConsumerV2.java

```diff
@@ -0,0 +1,515 @@
+            return self;
+        }
+
+        public Feature<Class<A>> apiClassFeature() {
+            // FIXME: reseting or disabling this feature should reset cached types feature
+            return apiClass;
+        }
+
+        public synchronized Builder<A> withFilterConfig(HollowFilterConfig filterConfig) {
+            requireNonNull(filterConfig, "filterConfig cannot be null; use filterConfigFeature().disable()");
+            return this.filterConfig.set(filterConfig);
+        }
+
+        public Feature<HollowFilterConfig> filterConfigFeature() {
+            return filterConfig;
+        }
+
+        public synchronized Builder<A> withDoubleSnapshotConfig(DoubleSnapshotConfig doubleSnapshotConfig) {
+            requireNonNull(doubleSnapshotConfig, "doubleSnapshotConfig cannot be null; use doubleSnapshotConfigFeature().disable()");
+            return this.doubleSnapshotConfig.set(doubleSnapshotConfig);
+        }
+
+        public Feature<DoubleSnapshotConfig> doubleSnapshotConfigFeature() {
+            return doubleSnapshotConfig;
+        }
+
+        public synchronized Builder<A> withObjectLongevityConfig(ObjectLongevityConfig objectLongevityConfig) {
+            requireNonNull(objectLongevityConfig, "objectLongevityConfig cannot be null; use objectLongevityConfigFeature().disable()");
+            return this.objectLongevityConfig.set(objectLongevityConfig);
+        }
+
+        public Feature<ObjectLongevityConfig> objectLongevityConfigFeature() {
+            return objectLongevityConfig;
+        }
+
+        public synchronized Builder<A> withObjectLongevityDetector(ObjectLongevityDetector objectLongevityDetector) {
+            requireNonNull(objectLongevityDetector, "objectLongevityDetector cannot be null; use objectLongevityDetectorFeature().disable()");
+            return this.objectLongevityDetector.set(objectLongevityDetector);
+        }
+
+        public Feature<ObjectLongevityDetector> objectLongevityDetectorFeature() {
+            return objectLongevityDetector;
+        }
+
+        public synchronized Builder<A> withRefreshExecutor(Executor refreshExecutor) {
+            requireNonNull(refreshExecutor, "refreshExecutor cannot be null; use noRefreshExecutor()");
+            return this.refreshExecutor.set(refreshExecutor);
+        }
+
+        public Feature<Executor> refreshExecutorFeature() {
+            return refreshExecutor;
+        }
+
+        @Deprecated
+        public synchronized Builder<A> withMetricsCollector(HollowMetricsCollector<HollowConsumerMetrics> metricsCollector) {
+            requireNonNull(metricsCollector, "metricsCollector cannot be null; use metricsCollectorFeature().disable()");
+            return this.metricsCollector.set(metricsCollector);
+        }
+
+        public Feature<HollowMetricsCollector<HollowConsumerMetrics>> metricsCollectorFeature() {
+            return metricsCollector;
+        }
+
+        public synchronized HollowConsumerV2<A> buildSync() {
```

Comment:
Not married to the naming of `buildSync` and `buildAsync` or even if the builder is responsible for kicking off the initial load. An alternative might be just be `HollowConsumer::start`, e.g. `consumer = builder.build().start()`.

Either way, we need to support an API where the initial load takes place in the caller's thread and not asynchronously with the caller blocking until ready (we want that use-case, too, hence `buildAsync()` or equivalent API sticks around).